mlir/include/mlir/Dialect/Linalg/Transforms/Transforms.h

```diff
@@ -110,6 +112,18 @@ Value bufferizeToAllocation(RewriterBase &rewriter,
                             vector::MaskOp maskOp, Attribute memorySpace = {},
                             Operation *insertionPoint = nullptr);
 
+/// Materialize a buffer allocation for the given bufferization.alloc_tensor op
+/// and lower the op to memref.alloc + memref.tensor_store.
+///
+/// In addition to rewriting the IR, this function returns the newly allocated
+/// buffer. The `insertionPoint` parameter can be used to specify a custom
+/// insertion point for the buffer allocation.
+Value bufferizeToAllocation(RewriterBase &rewriter,
+                            const BufferizeToAllocationOptions &options,
+                            bufferization::AllocTensorOp allocTensorOp,
+                            Attribute memorySpace = {},
+                            Operation *insertionPoint = nullptr);
```

Comment:
Should this be an actual `InsertionPoint` instead so it can insert into empty blocks? Also, can't it just use the insertion point configured for the `rewriter`?

Reply:
thanks for noting, since I am adding a 3rd case here I am complying to the existing APIs but let's track your comment in a bug for @matthias-springer to figure out globally